_drafts/GDB Tips and Tricks.md

```diff
@@ -0,0 +1,500 @@
+
+Once you extract it, you should see the following directory structure:
+
+
+![](Graphics/nRF5_SDK_folder.png)
+
+
+#### 2. nRF5 Command Line Tools
+Next, we need to install the nRF5 Command Line Tools. These include **nrfjprog**, which is a tool for programming your nRF52 development kit via the Segger J-Link debugger and needed in our case for working from the command line.
+
+**[Link to download the nRF5 Command Line Tools](https://www.nordicsemi.com/Software-and-Tools/Development-Tools/nRF5-Command-Line-Tools/Download#infotabs)**
+
+First, make sure you select the appropriate operating system (macOS in our case).
+
+
+![](Graphics/nRF5_Command_Line_Tools.png)
+<br/>
+
+Select the latest version, and then click "Download File".
+
+This is what the contents of the folder should look like:
+
+![](Graphics/nRF5_Command_Line_Tools_Folder.png)
+
+#### 3. SEGGER J-Link
+The SEGGER J-Link software is needed for the GDB Server interface to the nRF52 chipset on the development kit.
+
+So let's go ahead and download the software.
+
+[**Link to download SEGGER J-Link Software**](https://www.segger.com/downloads/jlink/#J-LinkSoftwareAndDocumentationPack)
+
+
+![](Graphics/J-Link_Software_Download.png)
+<br/>
+
+This download (for macOS) is a .**pkg** installer file. Once you download it, simply double-click it and go through the installation process.
+
+
+![](Graphics/J-Link_Installation.png)
+
+
+#### 4. GNU Arm Embedded Toolchain
+The next software package that we need to install is the GNU Arm Embedded Toolchain which includes the compiler (gcc) and debugger (gdb) for the Arm architecture (which the nRF52840 chipset is based on).
+
+**[Direct link to download the GNU Arm Embedded Toolchain (version 7-2017-q4-major)](https://developer.arm.com/-/media/Files/downloads/gnu-rm/7-2017q4/gcc-arm-none-eabi-7-2017-q4-major-mac.tar.bz2?revision=7f453378-b2c3-4c0d-8eab-e7d5db8ea32e?product=GNU%20Arm%20Embedded%20Toolchain,64-bit,,Mac%20OS%20X,7-2017-q4-major)**
+
+
+![](Graphics/GNU_Arm_Embedded_Download.png)
+<br/>
+
+After you download the package, simply extract it to your folder of choice.
+
+
+![](Graphics/GNU_Arm_Download_Folder.png)
+
+
+#### 5. Serial Terminal Program
+My choice for a terminal program that I use across projects is a program called **CoolTerm**. What I like about this program is its simplicity and that it supports all the major platforms (Windows, macOS, and Linux).
+
+**[Link to download CoolTerm](https://freeware.the-meiers.org/CoolTermMac.zip)**
+
+For now, all you need to do is download it, open the package and copy the application file to your Mac **Applications** folder so you could launch it in the later steps.
+
+#### 6. Adding Necessary Folders to $PATH
+In order to access the necessary commands that we'll be calling from anywhere in your system, you need to add their paths to the system **$PATH** environment variable.
+
+To do that, open up a Terminal session and type the following:
+
+```terminal
+$ sudo vi /etc/paths
```

Comment:
This is an odd choice to me. Why not update the path in `.bash_profile` instead?

```
$ vi ~/.bash_profile
...
export PATH = <Your Folder>/nRF-Command-Line-Tools_9_8_1_OSX/nrfjprog:$PATH
export PATH = <Your Folder>/gcc-arm-none-eabi-7-2017-q4-major/bin:$PATH
```

Reply:
Will update to reflect this.